atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/expectations/TrackingIteratorTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl.expectations;
+
+import static org.assertj.core.api.Assertions.assertThatIterator;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+import java.util.Iterator;
+import java.util.function.Consumer;
+import org.junit.Test;
+import org.mockito.InOrder;
+import org.mockito.Mockito;
+
+public class TrackingIteratorTest extends AbstractTrackingIteratorTest {
+    @Test
+    public void trackingIteratorForwardsData() {
+        TrackingIterator<String, Iterator<String>> trackingIterator =
+                new TrackingIterator<>(createStringIterator(), noOp(), STRING_MEASURER);
+        assertThatIterator(trackingIterator).toIterable().containsExactlyElementsOf(STRINGS);
+    }
+
+    @Test
+    public void trackerInvokedCorrectlyByTrackingIterator() {
+        Consumer<Long> tracker = spy(noOp());
+        InOrder inOrder = Mockito.inOrder(tracker);
+
+        TrackingIterator<String, Iterator<String>> trackingIterator =
+                new TrackingIterator<>(createStringIterator(), tracker, STRING_MEASURER);
+
+        trackingIterator.forEachRemaining(string -> {
+            inOrder.verify(tracker).accept(STRING_MEASURER.applyAsLong(string));
```

Comment:
I think this is fine, but wanted to flag as it's one of my design biases: I typically try to avoid verify unless I really have to (as it's a class out of our control, etc), what you could do here instead is simply have your track be `ArrayList::add`), and verify that the in-order list is equal to what you expect